### ISSUE TITLE
Fuse drops Drain Ok type?

### ISSUE THREAD

**Issue by Lucretiel** [slog-rs/slog#194]
Why does `Fuse` drop the `Ok` type of the underlying drain? Even though it panics on error, there doesn't seem to be a reason for it to always return `()` on success.

**Comment by dpc** [comment]
TL;DR: `Fuse` is mostly a convenience for users to adapt their `Drain` to fit https://docs.rs/slog/2.4.1/slog/struct.Logger.html#method.root (which requires `  D: 'static + SendSyncRefUnwindSafeDrain<Err = Never, Ok = ()>,`)


 The idea was that root `Logger` requires `Drain` that can not return any result anymore - good or bad. `Logger` has nothing to do with results, so this way we force the user to compose their `Drain` so that they have handled both the ok and errors. Then, for convenience `Fuse` was provided under the assumption, that most users will probably want to panic on errors. For `Fuse` to fit into root `Logger` constructor, it just ignores the `Ok` part.

If you don't want to ignore the `Ok` value, just write your own type. You can copy & paste `Fuse` from `slog` and modify it.
